tests/meltano/cli/test_cli.py

```diff
@@ -167,6 +169,58 @@ def test_handle_meltano_error(self):
             handle_meltano_error(exception)
 
 
+@pytest.mark.skipif(
+    platform.system() == "Windows",
+    reason="Windows terminal support for ANSI escape sequences is limited.",
+)
+class TestCliColors:
+    @pytest.mark.parametrize(
+        "no_color_flag,env,colors_expected",
+        [
+            pytest.param(
+                False,
+                {},
+                True,
+                id="colors-enabled-by-default",
+            ),
+            pytest.param(
+                True,
+                {},
+                False,
+                id="colors-disabled-by-no-color-flag",
+            ),
+            pytest.param(
+                False,
+                {
+                    "NO_COLOR": "1",
+                },
+                False,
+                id="colors-disabled-by-no-color-env",
+            ),
+        ],
+    )
+    def test_no_color(self, cli_runner, no_color_flag, env, colors_expected):
+        text = "This is a test"
+        styled_text = click.style(text, fg="red")
+
+        @cli.command("dummy")
+        @click.pass_context
+        def _dummy_command(ctx):
+            click.echo(styled_text)
+
+        args = ["dummy"]
+        expected_text = styled_text if colors_expected else text
+
+        if no_color_flag:
+            args = ["--no-color"] + args
+
+        with cli_runner.isolated_filesystem():
```

Comment:
The random order meant that some other test was leaving an empty `meltano.yml` that was causing problems. [`isolated_filesystem`](https://click.palletsprojects.com/en/8.1.x/testing/#file-system-isolation) just ensures  the CLI is invoked in an empty directory.